commands/audit/audit.go

```diff
@@ -202,7 +202,7 @@ func RunAudit(auditParams *AuditParams) (results *utils.Results, err error) {
 	}
 	// The sca scan doesn't require the analyzer manager, so it can run separately from the analyzer manager download routine.
 	if scaScanErr := buildDepTreeAndRunScaScan(auditParallelRunner, auditParams, results); scaScanErr != nil {
-		auditParallelRunner.AddErrorToChan(scaScanErr)
+		_ = createErrorIfPartialResultsDisabled(auditParams, auditParallelRunner, fmt.Sprintf("An error has occurred during SCA scan process. SCA scan is skipped for the following directories: %s.", auditParams.workingDirs), scaScanErr)
```

Comment:
```suggestion
		err = createErrorIfPartialResultsDisabled(auditParams, auditParallelRunner, "An error has occurred during SCA scan process.", scaScanErr)
```
no need to specify the working dirs here, the error for the specific scan will handle adding its information to the error. In addition, you are not doing anything with the returned `error`, you should handle it or change the method

Reply:
I can remove the working dirs from the error message
But it doesn't make since to return the error since we are doing nothing with this error. with AddErrorToChan we just added the error to the runner errors queue and this is what is done here as well. To avoid code duplication with writing the exact same logic twice - one with returning the error and one without I put it all into a single function and explained exactly how it works:
if a parallelRunner if provided the error is getting added to the error queue, if not - the error is returned.
In this case we only add it to the queue and it is later being handled
@attiasas